### PR TITLE
Updated C# code gen to read both content and message headers when generating a Header object

### PR DESCRIPTION
**Pull request by stankovski**

### CODE REVIEW

File: ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime/HttpExtensions.cs [hunk 1]
```diff
@@ -164,5 +164,46 @@ public static JObject ToJson(this HttpResponseHeaders headers)
...
+        /// <returns>Json string</returns>
+        public static JObject GetHeadersAsJson(this HttpResponseMessage message)
+        {
+            if (message == null)
+            {
+                return new JObject();
+            }
+
+            var jObject = new JObject();
+            foreach (var httpResponseHeader in message.Headers)
```
Comment (matthchr) [inline ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime/HttpExtensions.cs]: Can message.Headers be null?

File: ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime.Tests/JsonSerializationTests.cs [hunk 2]
```diff
@@ -262,5 +263,21 @@ public void HeaderGetsSerializedToJson()
                 Environment.NewLine);
             Assert.Equal(expectedJsonString, json);
         }
+
+        [Fact]
+        public void HeaderKnownValuesGetSerializedToJson()
+        {
+            var message = new HttpResponseMessage();
+            message.Content = new StringContent("Test");
```
Comment (matthchr) [inline ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime.Tests/JsonSerializationTests.cs]: Can you please get Content-Length or Content-Type (a "well-named" header)?

File: ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime/HttpExtensions.cs (hunk 1 above)
Comment (stankovski) [inline ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime/HttpExtensions.cs]: It's not really documented anywhere but based on the source code it should never be null - https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs#L99

**Comment by matthchr** [comment]
@stankovski This looks good to me

**Pull request merged**